services/web/server/src/simcore_service_webserver/projects/projects_service.py

```diff
@@ -971,13 +975,23 @@ async def update_project_node_state(
         permission="write",  # NOTE: MD: before only read was sufficient, double check this
     )
 
+    # TODO: https://github.com/ITISFoundation/osparc-simcore/issues/7046
+    # delete this once workbench is removed from the projects table
     updated_project, _ = await db.update_project_node_data(
         user_id=user_id,
         project_uuid=project_id,
         node_id=node_id,
         product_name=None,
         new_node_data={"state": {"currentStatus": new_state}},
     )
+
+    await _projects_nodes_repository.update(
```

Comment:
I guess is not the right moment (since we have to update projects repo as well), but I guess calling all this repos to update should be done in a single transaction, right?  i.e.

```python
async with transaction_context(...) as conn:
    # ... 
    await _projects_repository.update_project_node_data(app, conn, ...)
    await _projects_nodes_repository.update(app, conn, 
```